common/src/app/common/types/container.cljc

```diff
@@ -413,15 +413,10 @@
   (let [children (cfh/get-children-with-self objects (:id shape))
         parents  (cfh/get-parents objects (:id shape))]
     (or
+     (ctk/main-instance? shape)
```

Comment:
This seems redundant, isn't it? children includes self, so if shape is main it should be detected by the following line

Reply:
Oh, that's correct!